plugins/destination/duckdb/client/transform_gotype.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"time"
+
+	"github.com/apache/arrow/go/v15/arrow"
+	"github.com/apache/arrow/go/v15/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v4/types"
+	"github.com/marcboeker/go-duckdb"
+)
+
+func getValue(arr arrow.Array, i int) any {
+	if !arr.IsValid(i) {
+		if arrow.TypeEqual(arr.DataType(), types.NewUUIDType()) {
+			return duckdb.UUID{}
+		}
+
+		// return typed nil
+		switch arr.DataType().ID() {
+		case arrow.TIMESTAMP:
+			return ptrOf[time.Time]()
+		case arrow.BOOL:
+			return ptrOf[bool]()
+		case arrow.INT8:
+			return ptrOf[int8]()
+		case arrow.INT16:
+			return ptrOf[int16]()
+		case arrow.INT32:
+			return ptrOf[int32]()
+		case arrow.INT64:
+			return ptrOf[int64]()
+		case arrow.UINT8:
+			return ptrOf[uint8]()
+		case arrow.UINT16:
+			return ptrOf[uint16]()
+		case arrow.UINT32:
+			return ptrOf[uint32]()
+		case arrow.UINT64:
+			return ptrOf[uint64]()
+		case arrow.FLOAT32:
+			return ptrOf[float32]()
+		case arrow.FLOAT64:
+			return ptrOf[float64]()
+		case arrow.STRING:
+			return ptrOf[string]()
+		case arrow.BINARY, arrow.LARGE_BINARY, arrow.FIXED_SIZE_BINARY:
+			return ptrOf[[]byte]()
+		default:
+			return ptrOf[string]()
+		}
+	}
+
+	if arrow.TypeEqual(arr.DataType(), types.NewUUIDType()) {
```

Comment:
This is probably ineffective

Reply:
can be incorporated with the type switch as `case *types.UUIDType:`